12:18:58 - PyTorch version 2.5.1 available.
[93mWARNING [0m:   DEPRECATED FEATURE: flwr.server.start_server() is deprecated.
	Instead, use the `flower-superlink` CLI command to start a SuperLink as shown below:

		$ flower-superlink --insecure

	To view usage and all available options, run:

		$ flower-superlink --help

	Using `start_server()` is deprecated.

            This is a deprecated feature. It will be removed
            entirely in future versions of Flower.
        
12:19:09 - DEPRECATED FEATURE: flwr.server.start_server() is deprecated.
	Instead, use the `flower-superlink` CLI command to start a SuperLink as shown below:

		$ flower-superlink --insecure

	To view usage and all available options, run:

		$ flower-superlink --help

	Using `start_server()` is deprecated.

            This is a deprecated feature. It will be removed
            entirely in future versions of Flower.
        
[92mINFO [0m:      Starting Flower server, config: num_rounds=100, no round_timeout
12:19:09 - Starting Flower server, config: num_rounds=100, no round_timeout
[92mINFO [0m:      Flower ECE: gRPC server running (100 rounds), SSL is disabled
12:19:09 - Flower ECE: gRPC server running (100 rounds), SSL is disabled
[92mINFO [0m:      [INIT]
12:19:09 - [INIT]
[92mINFO [0m:      Using initial global parameters provided by strategy
12:19:09 - Using initial global parameters provided by strategy
[92mINFO [0m:      Starting evaluation of initial global parameters
12:19:09 - Starting evaluation of initial global parameters
[92mINFO [0m:      initial parameters (loss, other metrics): 0.6899258938490176, {'accuracy': 0.5392156862745098, 'f1': 0.584070796460177, 'testing_loss': 0.6899258938490176}
12:19:10 - initial parameters (loss, other metrics): 0.6899258938490176, {'accuracy': 0.5392156862745098, 'f1': 0.584070796460177, 'testing_loss': 0.6899258938490176}
[92mINFO [0m:      
12:19:10 - 
[92mINFO [0m:      [ROUND 1]
12:19:10 - [ROUND 1]
[92mINFO [0m:      configure_fit: strategy sampled 2 clients (out of 2)
12:19:14 - configure_fit: strategy sampled 2 clients (out of 2)
12:19:19 - [Monitor Variance] Aggregated local states successfully from Client IDs [1, 0]!
12:19:19 - [Monitor Variance] Successfully sent back to Client 1 at localhost:9201 the variance approximation 0.0061!
12:19:19 - [Monitor Variance] Successfully sent back to Client 0 at localhost:9199 the variance approximation 0.0061!
[92mINFO [0m:      aggregate_fit: received 2 results and 0 failures
12:19:19 - aggregate_fit: received 2 results and 0 failures
Traceback (most recent call last):
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/mtheologitis/FDA-Opt-Sys/fdaopt/server.py", line 124, in <module>
    start_server(
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/site-packages/flwr/server/app.py", line 230, in start_server
    hist = run_fl(
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/site-packages/flwr/server/server.py", line 492, in run_fl
    hist, elapsed_time = server.fit(
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/site-packages/flwr/server/server.py", line 115, in fit
    res_fit = self.fit_round(
  File "/home/mtheologitis/anaconda3/envs/flower/lib/python3.10/site-packages/flwr/server/server.py", line 251, in fit_round
    ] = self.strategy.aggregate_fit(server_round, results, failures)
  File "/home/mtheologitis/FDA-Opt-Sys/fdaopt/fda_strategies.py", line 55, in aggregate_fit
    client_vecs = [
  File "/home/mtheologitis/FDA-Opt-Sys/fdaopt/fda_strategies.py", line 56, in <listcomp>
    vectorize_ndarrays(client_params, self.device)
AttributeError: 'FdaAdam' object has no attribute 'device'
